src/PID/MotorControllerEnhanced.java

```diff
@@ -0,0 +1,73 @@
+package PID;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.IMotorControllerEnhanced;
+
+public class MotorControllerEnhanced implements PIDController {
+  private IMotorControllerEnhanced motorControllerEnhanced;
+  private PIDFTerms pidfTerms;
+
+  public MotorControllerEnhanced(IMotorControllerEnhanced motorControllerEnhanced) {
```

Comment:
I see no reason to create a PID controller with no setpoint. So include it in all constructors.